server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/OperationHandler.java

```diff
@@ -0,0 +1,242 @@
+
+            CFNCommand cfnCommand = handleOperationCommand(artifact, inputs);
+            markFile(artifact);
+            CFNFile cfnFile = handleOperationFile(artifact, MODE_500, serverName);
+
+            // Add file to config and execution command
+            cfnModule.getCFNInit(serverName)
+                .getOrAddConfig(CONFIG_SETS, config)
+                .putFile(cfnFile)
+                .putCommand(cfnCommand);
+        }
+    }
+
+    /**
+     Takes an artifact path and input variables and returns the corresponding CloudFormation command with input variables.
+
+     @param artifact path to the artifact
+     @param inputs   set with all input variables
+     @return CFNCommand corresponding to the given artifact
+     */
+    private CFNCommand handleOperationCommand(String artifact, Set<OperationVariable> inputs) {
+        String parent = new File(artifact).getParent();
+        if (parent == null) {
+            parent = "";
+        }
+        CFNCommand cfnCommand = new CFNCommand(artifact,
+            ABSOLUTE_FILE_PATH + artifact) //file is the full path, so need for "./"
+            .setCwd(ABSOLUTE_FILE_PATH + parent);
+        // add inputs to environment
+        for (OperationVariable input : inputs) {
+            String value = input.getValue().orElseThrow(
+                () -> new IllegalArgumentException("Input value of " + input.getKey() + " expected to not be " +
+                    "null")
+            );
+            if (cfnModule.checkFn(value)) {
+                cfnCommand.addEnv(input.getKey(), cfnModule.getFn(value));
+            } else {
+                cfnCommand.addEnv(input.getKey(), value);
+            }
+        }
+        return cfnCommand;
+    }
+
+    /**
+     Takes a filePath, mode and instance name and returns the corresponding CloudFormation file.
+     Also marks instance as in need of authentication and file to be uploaded if needed.
+
+     @param filePath   path to the file
+     @param mode       mode to set access rights
+     @param serverName corresponding to the EC2 instance
+     @return CFNFile corresponding to the given filePath, mode and serverName
+     */
+    private CFNFile handleOperationFile(String filePath, String mode, String serverName) {
+        String cfnSource = getFileURL(cfnModule.getBucketName(), filePath);
+
+        if (!cfnModule.getAuthenticationSet().contains(serverName)) {
+            logger.debug("Marking '{}' as instance in need of authentication.", serverName);
+        } else {
+            logger.debug("'{}' already marked as instance in need of authentication. " +
+                "Skipping authentication marking.", serverName);
+        }
+        cfnModule.putAuthentication(serverName);
+
+        return new CFNFile(ABSOLUTE_FILE_PATH + filePath)
+            .setSource(cfnSource)
+            .setMode(mode) //TODO Check what mode is needed (read? + execute?)
+            .setOwner(OWNER_GROUP_ROOT) //TODO Check what Owner is needed
```

Comment:
Are these todos still a thing? please consider opening issues for them if they wont be resolved in this PR.

Reply:
See #406.